docs/scarpet/api/Inventories.md

```diff
@@ -206,3 +206,138 @@ Returns size of the actual dropped items.
 inventory_drop(player(), 0, 1) => 1 // Q's one item on the ground
 inventory_drop(x,y,z, 0) => 64 // removed and spawned in the world a full stack of items
 </pre>
+
+## Screen handlers
+
+A screen handler is a value type used to open screens to a player and interact with them.
+For example, this includes the chest inventory gui, the crafting table gui and many more.
+
+### `create_screen(player, type, name, callback?)`
+
+Creates and opens a screen to a `player`.
+
+Available `type`s:
+
+* `anvil`
+* `beacon`
+* `brewing_stand`
+* `cartography_table`
+* `crafting`
+* `enchantment`
+* `furnace`
+* `generic_3x3`
+* `generic_9x1`
+* `generic_9x2`
+* `generic_9x3`
+* `generic_9x4`
+* `generic_9x5`
+* `generic_9x6`
+* `grindstone`
+* `hopper`
+* `lectern`
+* `loom`
+* `merchant`
+* `shulker_box`
+* `smithing`
+* `stonecutter`
+
+The `name` parameter can be a formatted text and will be displayed at the top of the screen.
+Some screens like the lectern or beacon screen don't show it.
+
+Optionally, a `callback` function can be passed as a fourth argument.
+This functions needs to have five parameters:
+`_(screen, player, action, index, button) -> ...`
+
+The `screen` parameter is the screen handler value of the screen itself.
+`player` is the player who interacted with the screen.
+`action` is a string corresponding to the interaction type.
+Can be one of:
+
+Slot interactions:
+* `pickup`
+* `quick_move`
+* `swap`
+* `clone`
+* `throw`
+* `quick_craft`
+* `pickup_all`
+
+Other interactions:
+* `button` Pressing a button in certain screens that have then (enchantment table, lectern, loom and stonecutter)
+* `close` Triggers when the screen gets closed.
+
+`index` is the slot index of the slot interaction.
+If clicked outside the screen (where it would drop held items), this value is -999.
+In the case of a button interaction, this is the button index.
+Note that for lecterns, this index can be certain a value above 100, for jumping to a certain page.
+This can come from formatted text inside the book, with a `change_page` click event action.
+
+`button` corresponds to the mouse button used for the slot click event.
+This only works for slot click interactions, for `button` and `close` type, this value is 0.
+
+0 -> left click
+1 -> right click
+2 -> middle click
+3, 4 -> extra mouse buttons (don't work reliably, only if holding an item in the cursor slot)
+
+By returning a string `'cancel'` in the callback function,
+the screen interaction can be cancelled.
+This doesn't work for the `close` action.
+
+The `create_screen` function returns a `screen_handler` value,
```

Comment:
suggest simplifying from `screen_handler` to `screen`. Technically everything is a handler, `entity_handler`, `number_handler` etc.